security-analysis/security-analysis-api/src/main/java/com/powsybl/security/action/GeneratorAction.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.security.action;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * An action to:
+ * - change the targetP of a generator, either by specifying a new absolute value (MW) or a relative change (MW).
+ * - enable or disable the generator voltage control.
+ * - change targetQ of a generator (MVar).
+ * - change targetV of a generator (kV).
+ *
+ * @author Hadrien Godard <hadrien.godard@artelys.com>
+ * @author Anne Tilloy <anne.tilloy@rte-france.com>
+ */
+public class GeneratorAction extends AbstractAction {
+
+    public static final String NAME = "GENERATOR";
+
+    private final String generatorId;
+    private final Boolean activePowerRelativeValue; // true if it is a relative variation of targetP, false if it is a new targetP
```

Comment:
the comments would have more value as javadoc (note that it's even more important for those classe which do not implement the behaviour, but only specify them)